reconstruct/reconstruct.go

```diff
@@ -101,9 +105,95 @@ func ReconAccuracy(simFilename string, reconFilename string, leavesOnlyFile stri
 	acc := 100 - accuracy
 	answer[allNodes] = acc
 
+	return answer, accByBase
+}
+
+// ReconAccuracyByBase will run if the option given to ReconAccuracy = 1. This function calculates the percentage of first, second and third bases in codons that were correct.
+func ReconAccuracyByBase(simFilename string, reconFilename string, gpFilename string) map[string][]float64 {
+	sim := fasta.Read(simFilename)
+	recon := fasta.Read(reconFilename)
+	genes := genePred.Read(gpFilename)
+	answer := make(map[string][]float64)
+
+	for s := 0; s < len(sim); s++ {
+		for r := 0; r < len(recon); r++ {
+			var percentage1 float64
+			var percentage2 float64
+			var percentage3 float64
+			var mistakes1 float64
+			var mistakes2 float64
+			var mistakes3 float64
+			var total1 float64
+			var total2 float64
+			var total3 float64
+			if sim[s].Name == recon[r].Name {
+				for i := 0; i < len(sim[s].Seq); i++ {
+					for g := 0; g < len(genes); g++ {
+						inExon, exon := simulate.CheckExon(genes[g], i)
+						if inExon {
+							loc := findLocationInCodon(genes[g], exon, i)
+							if loc == 1 {
```

Comment:
Feel free to disagree, but this is the exact situation switch cases were designed for. Consider changing to 
```
switch loc {
case 1: // some work
case 2: // some work
case 3: // some work
}
```